### PR TITLE
OSDOCS-3301:Adding web console secrets

### PR DESCRIPTION
**Pull request by mpandya15**
[OSDOCS-3301]: Adding info on creating secrets using the web console

Version(s):

4.11+
Issue:
https://issues.redhat.com/browse/OSDOCS-3301

Link to docs preview:
https://62980--docspreview.netlify.app/openshift-enterprise/latest/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-creating-web-console-secrets.adoc_nodes-pods-secrets

QE review:
- [X] QE has approved this change.

Additional information:
The JIRA is divided into two sub-tasks. This PR addresses sub task 1.

### CODE REVIEW

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (mpandya15) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:13]: Would it be helpful to insert an alternative sample YAML here for clarity?

Comment (mburke5678) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:13]: @mpandya15 Yes. I think it would be helpful to add an example YAML. It would provide consistency with the other sections, each of which have an example.  Maybe list the secret types and point to the following sections?

Comment (mburke5678) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:13]: Also, maybe swap your new module with Creating opaque secrets, so that all of the secret type modules are together?

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (mburke5678) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:16]: s/drop down/drop-down

**Comment by mburke5678** [comment]
@mpandya15 A few thoughts and one small change. Otherwise LGTM!

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (skrthomas) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:35]:
```suggestion
The following sections describe how to create an opaque secret, a service account token secret, a basic authentication secret, an SSH authentication secret, and a Docker configuration secret using CLI. You can alternatively use the web console to create these secrets.   
```

Comment (skrthomas) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:35]: One small suggestion :)

Comment (skrthomas) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:35]: Less certain about the "Alternatively use the web console to create these secrets" bit.

Comment (mpandya15) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:35]: Thanks Sara! I added the "CLI" bit and held off on the "Alternatively use the web console to create these secrets" bit.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (mpandya15) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc]: @mburke5678 Thanks a lot for your feedback! I added the sample YAML, but is there another custom, Secret YAML example I can use for this, since we already have an example of Opaque? Opaque is the default in the console but maybe something different would be better.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (bergerhoffer) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:10]: You need a `.Procedure` heading here!

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (bergerhoffer) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:7]: Our procedures should start with a gerund ("Creating").

I'd also recommend just saying "secret" instead of "Secret object", to be consistent with the other procedures

So: 

```
= Creating a secret by using the web console
```

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (bergerhoffer) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:6]: The `.adoc` shouldn't be part of the ID. So:

```
[id="nodes-pods-secrets-creating-web-console-secrets_{context}"]
```

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (bergerhoffer) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:35]: This module could be reused in the future, perhaps not with those following sections. So I would recommend just removing this line

File: nodes/pods/nodes-pods-secrets.adoc
Comment (bergerhoffer) [inline nodes/pods/nodes-pods-secrets.adoc:30]: This section might make more sense to come after all of the other existing CLI procedures. Since you don't clarify in the titles that those are all CLI, after you've said web console for the first one. (And I don't think that's necessary, but reordering should be enough)

So I think it would make sense to show all of those existing ways to create the various types of secrets by using the CLI. And then at the end, it's like oh yeah, here's how to create those secrets through the web console, if you wanted to do that instead.

So I'd consider moving it down to after "include::modules/nodes-pods-secrets-creating-docker.adoc[leveloffset=+2]"

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (bergerhoffer) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:11]: Instead of using the icon, you should use "->". So:

```
. Navigate to *Workloads* -> *Secrets*.
```

Same for the next line

**Review by bergerhoffer — COMMENTED** [review]
A few additional suggestions!

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (mpandya15) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:35]: Removed this last sentence.

**Comment by xingxingxia** [comment]
@HanielZx please help review.

File: nodes/pods/nodes-pods-secrets.adoc
Comment (HanielZx) [inline nodes/pods/nodes-pods-secrets.adoc:67]: There are one typography issue as follow picture. In the GitHub file change, I see that "Additional resources" is before article "Creating a secret using the web console", but in fact the "Additional resources" is on the first paragraph of article.
And refer to other part, I think the "Additional resources" should not put on the first paragraph, it may put before "For more information ...".
截屏2023-08-04 10 55 52

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (HanielZx) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:7]: There are three demands in the [OSDOCS-3301](https://issues.redhat.com/browse/OSDOCS-3301), but I think that this file only meet "As a user, I want to add secrets from the web console". These two need to be added:
- As a user, I want to manage secrets from the web console. 
- As a user, I want to add secrets as environment variables (via the web console and CLI), such as https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables

**Review by HanielZx — CHANGES_REQUESTED** [review]

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (stlaz) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:24]: I don't want the default namespace to appear in examples as it's a rather special snowflake among namespaces.

Plaes use a made-up name for the namespace if it needs to be specified

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (stlaz) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:30]: This reads as `opaque means opaque` :slightly_smiling_face: 

Try something more helpful, you can mention other secret types, for example.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (stlaz) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:9]: A user does not necessarily have to be a namespace/cluster administrator to create secrets.

**Review by stlaz — CHANGES_REQUESTED** [review]

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (stlaz) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:26]: `stringData` is actually not what people typically use secrets for but it's worth mentioning.
Add `data`

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (mpandya15) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:7]: Thank you bringing this up. The second demand about adding [secrets as environment variables](https://docs.openshift.com/container-platform/4.13/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-creating_nodes-pods-secrets) is already addressed in example with the title "YAML of a pod populating environment variables with secret data", so that task will be removed from the to-do. I will work on the managing secrets portion.

**Comment by stlaz** [comment]
lgtm, I wonder if @jhadvig would like to chime in on the console parts

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by mpandya15** [comment]
@HanielZx Could you please review and approve? Thanks!

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (xingxingxia) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:24]: The indentation is only one space. Inconsistent with normally even number of spaces.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (xingxingxia) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:27]: This is incorrect. In this YAML example, `data` isn't given any values at all.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (xingxingxia) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:32]: Is line 32 <2> intended for explaining `stringData: <2>`?  Line 32 does not explain usage of `stringData: <2>`. If intended for explaining `stringData: <2>`, you could refer to the statement `Entries in the stringData map are converted to base64 and the entry will then be moved to the data ...` under the `Understanding secrets` section.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (xingxingxia) [reply modules/nodes-pods-secrets-creating-web-console-secrets.adoc:27]: The `Understanding secrets` section has example of `data` and `stringData` usage.

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (skrthomas) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:33]:
```suggestion
<2> Entries in the `stringData` map are converted to base64 and the entry will then be moved to the `data` map automatically. This field is write-only; the value will only be returned via the `data` field.
```

File: modules/nodes-pods-secrets-creating-web-console-secrets.adoc
Comment (xingxingxia) [inline modules/nodes-pods-secrets-creating-web-console-secrets.adoc:30]: Some nits, otherwise LGTM:
- indent with two spaces for each newly nested filed. Four spaces work but not conventional.
- Under `data`, the fields' values are base64 encoded instead of raw values.
- `my app` has a typo of an unneeded space

```suggestion
metadata:
  name: example
  namespace: <namespace>
type: Opaque <1>
data:
  username: <base64 encoded username>
  password: <base64 encoded password>
stringData: <2>
  hostname: myapp.mydomain.com 
```

**Comment by skrthomas** [comment]
@sagidlow Hey I just noticed this PR never got merged for Manushri after her internship. I can take it up and get it through merge review. Just wanted to check with you about it first.

**Comment by jab-rh** [comment]
@skrthomas, do we know what versions this applies to? Thanks!

**Comment by skrthomas** [comment]
@jab-rh sorry about that, I added the versions.

**Comment by xingxingxia** [comment]
@skrthomas @jab-rh this PR does not get approved with /lgtm label, it has incorrect content https://github.com/openshift/openshift-docs/pull/62980#pullrequestreview-1574803578 . Please open new PR to update that! Thanks!!

**Comment by skrthomas** [comment]
@xingxingxia I will open a new PR and fix this unaddressed comment ASAP and I'll add you for review. TY, I had removed the merge-review-needed label but I think mine and @jab-rh wires were crossed.

**Pull request merged**
